e2etests/testing.go

```diff
@@ -120,7 +120,7 @@ func (tc *TestCluster) initialize() error {
 		err := runCmd(
 			"go",
 			[]string{"build", "-o", "../hcloud-cloud-controller-manager", "../."},
-			[]string{"CGO_ENABLED=0"},
+			[]string{"CGO_ENABLED=0", "GOOS=linux", "GOARCH=amd64"},
```

Comment:
Why do we need this change?

Reply:
@fhofherr added this a few weeks ago, until then I didn't run our E2E tests in the CCM, because otherwise I would noticed earlier that it builds a macos binary and tries to run it on a ubuntu server :) For @fhofherr it worked (out of the box) because he uses Linux amd64 as his development machine.